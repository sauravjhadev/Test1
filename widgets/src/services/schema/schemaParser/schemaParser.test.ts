import fs from 'fs';

import { generateWidgetConfiguration } from './schemaParser';

const schemaData = fs.readFileSync('src/services/__fixtures__/schema.json', 'utf8').toString();

describe('generateWidgetConfiguration', () => {
    it('correctly generates a widget configuration', () => {
        expect(generateWidgetConfiguration(JSON.parse(schemaData))).toMatchSnapshot();
    });
});
