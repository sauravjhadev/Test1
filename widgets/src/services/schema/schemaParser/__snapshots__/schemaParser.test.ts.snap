// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateWidgetConfiguration correctly generates a widget configuration 1`] = `
Object {
  "animationDuration": "400",
  "buttonActionType": "addToCart",
  "buttonBorderColor": "rgba(102,102,102,1);",
  "buttonBorderColorHover": "rgba(102,102,102,1);",
  "buttonBorderRadius": 4,
  "buttonColor": "rgba(68,68,68,1)",
  "buttonColorHover": "rgba(102,102,102,1);",
  "buttonFontFamily": "inherit",
  "buttonFontSize": "14",
  "buttonFontWeight": "400",
  "buttonLabel": "Add to Cart",
  "buttonPseudo": "normal",
  "buttonTextColor": "rgba(255,255,255,1)",
  "buttonTextColorHover": "rgba(255,255,255,1)",
  "buttonTextDecoration": "none",
  "buttonTextDecorationHover": "none",
  "carouselArrowBackgroundColor": "rgba(255,255,255,0.0)",
  "carouselArrowBorderColor": "rgba(255,255,255,0.0)",
  "carouselArrowColor": "#D9DCE9",
  "draggable": true,
  "easingMethod": "ease",
  "productCardBackgroundColor": "rgba(255,255,255,1)",
  "productCardBorderColor": "#D9DCE9",
  "productCardContentAlignment": Object {
    "horizontal": "center",
  },
  "productCardMargin": Object {
    "bottom": Object {
      "type": "px",
      "value": "0",
    },
    "left": Object {
      "type": "px",
      "value": "10.5",
    },
    "right": Object {
      "type": "px",
      "value": "10.5",
    },
    "top": Object {
      "type": "px",
      "value": "0",
    },
  },
  "productCardPadding": Object {
    "bottom": Object {
      "type": "px",
      "value": "16",
    },
    "left": Object {
      "type": "px",
      "value": "0",
    },
    "right": Object {
      "type": "px",
      "value": "0",
    },
    "top": Object {
      "type": "px",
      "value": "0",
    },
  },
  "productDescriptionFontFamily": "inherit",
  "productDescriptionFontSize": "14",
  "productDescriptionFontWeight": "400",
  "productDescriptionLineHeight": "21",
  "productDescriptionTextColor": "#000000",
  "productImageHeight": "252",
  "productNameFontFamily": "inherit",
  "productNameFontSize": "22",
  "productNameFontWeight": "400",
  "productNameLineHeight": "33",
  "productNamePseudo": "normal",
  "productNameTextColor": "#000000",
  "productNameTextColorHover": "#000000",
  "productNameTextDecoration": "none",
  "productNameTextDecorationHover": "none",
  "productPriceFontFamily": "inherit",
  "productPriceFontSize": "14",
  "productPriceFontWeight": "400",
  "productPriceLineHeight": "21",
  "productPriceTextColor": "#000000",
  "productRatingStarColorEmpty": "#D9DCE9",
  "productRatingStarColorFull": "#80000B",
  "productRatingsTextColor": "#8C93AD",
  "products": Array [
    Object {
      "productImageAlignment": Object {
        "horizontal": "center",
        "vertical": "middle",
      },
      "productImageFit": "fit",
    },
    Object {
      "productImageAlignment": Object {
        "horizontal": "center",
        "vertical": "middle",
      },
      "productImageFit": "fit",
    },
    Object {
      "productImageAlignment": Object {
        "horizontal": "center",
        "vertical": "middle",
      },
      "productImageFit": "fit",
    },
  ],
  "productsPerRow": "3",
  "showButton": true,
  "showProductDescription": true,
  "showProductName": true,
  "showProductPrice": true,
  "showProductRatings": true,
  "subtitle": "Description text goes here",
  "title": "Sample title text",
}
`;
